include/sound/sof/topology.h

```diff
@@ -300,4 +300,9 @@ enum sof_event_types {
 	SOF_KEYWORD_DETECT_DAPM_EVENT,
 };
 
+/* extended data struct for component new */
+struct sof_ipc_comp_new_ext {
+	uint8_t uuid[16];
+}  __packed;
```

Comment:
> sizeof(uint8_t) > sizeof(char)

This would be legal (and crazy) but this sort of "inner padding" would be unrelated to alignment and `__packed` would have no effect on it anyway.

Now assuming `sizeof(uint8_t)` is (sanely) equal to 1, the only way I think `__packed` could maybe make a difference is if some (also crazy?) compiler would like to align all structs on 32 bytes boundaries for some strange performance reason. I don't think you would care about that so please remove `__packed` here, it's confusing. I guess it's just left over from some previous, more complex structure?

Reply:
@marc-hb I think the reason why we added __packed to all the IPC structs is wanted to make sure the there is no padding among members of the struct. The extended data struct now is only 16 uint8_t, but it will be expended soon and will become more and more complicate. So assuming there is no harm with the __packed here, we can keep it and make all IPC struct in the consistent style?